src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+import duke.exceptions.InvalidStartEndException;
+import duke.tasks.*;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * deals with loading tasks from the file, and saving tasks in the file
+ */
+public class Storage {
+    private String path;
+    private Ui ui;
+    public Storage(String path, Ui ui) {
+        this.path = path;
+    }
+    public static LocalDateTime stringToDateTime(String str) throws DateTimeParseException {
+        //check if dateTime has correct format: ie. YYYY-MM-DD 00:00
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        LocalDateTime dateTime = LocalDateTime.parse(str, formatter);
+        return dateTime;
+    }
+    public File createDataFile() {
+        File dataFile = new File(this.path);
+        try {
+            Path dirPath = Paths.get("./data/");
+            if (!Files.exists(dirPath)) {
+                Files.createDirectory(dirPath);
+            }
+            if (!dataFile.exists()) {
+                dataFile.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("Unable to create file!!");
+        }
+        return dataFile;
+    }
+
+    public ArrayList<Task> loadTasks() {
```

Comment:
personally i feel that Storage should only be responsible for reading what is stored in the file. Maybe the logic for converting the text to the Task could be moved to parser or somewhere else relevant. What do you think?